src/Halogen/Component.purs

```diff
@@ -147,7 +72,41 @@ component = mkComponent <<< go
 type ComponentSpec h s f act ps i o m =
   { initialState :: i -> s
   , render :: s -> h (ComponentSlot h ps m act) act
-  , eval :: HalogenQ f act i ~> HalogenM' s act ps o m
+  , eval :: HalogenQ f act i ~> HalogenM s act ps o m
+  }
+
+type EvalSpec s f act ps i o m =
+  { handleAction :: act -> HalogenM s act ps o m Unit
+  , handleQuery :: forall a. f a -> HalogenM s act ps o m (Maybe a)
+  , receive :: i -> Maybe act
+  , initialize :: Maybe act
+  , finalize :: Maybe act
+  }
+
+mkEval
```

Comment:
What you think about calling `mkEval` from `mkComponent` (i.e. `ComponentSpec` taking `EvalSpec` ?

Reply:
We can't do that without exposing another different version of `mkComponent`, since Halogen uses it internally. Plus I wanted to provide the choice of writing an `eval` in terms of `HalogenQ` directly. 

Having `mkEval` in there seemed better given both of those considerations, as it should avoid some confusion about things like when we had 4 different component constructors before - they were all the same aside from some synonyms and field-defaulting, but that wasn't obvious to the casual user. Since the same actual `component` constructor is used here, and we just use another function to build up part of its internals, it shows that there's nothing special going on, no matter how the component is constructed.